### PR TITLE
all: add debug logs to Go tracer/integrations

### LINKED ISSUE #653: proposal: add additional debug logging to tracer and integrations
Debug logging is useful when diagnosing an issue with an application, and we should consider adding additional information to our logs.

Potentially useful info to include in debug logs:
- [x] complete tracer config on tracer start
- [x] show trace id, span ids, operation name, span tags for each span generated - helps us find out if the tracer generated a specific span
- [ ] log integration configurations when they are successfully attached/started
- [ ] log any integration errors preventing the integration from working (do we want to do this? We already return `error`s when something fails)
- [x] for distributed tracing, have a debug log line if we’ve extracted trace-id.

### PR DESCRIPTION
**Pull request by nicoledanuwidjaja**
Continuation of [this PR](https://github.com/DataDog/dd-trace-go/pull/677) by @knusbaum. Updates debug logs for newer integrations at config startup. 

Open to discussion for now to see if any parts of the issue are unaddressed.

Fixes #653

### CODE REVIEW

**Comment by nicoledanuwidjaja** [comment]
The `go-pg/pg.v10` integration doesn't have a config; should a debug log be added to the database wrapper?

**Comment by knusbaum** [comment]
@nicoledanuwidjaja I think we should add a line to the function here:
https://github.com/DataDog/dd-trace-go/blob/3ed2a2784fd7adb414f2f47dbb6b6a1d301cdf8c/contrib/go-pg/pg.v10/pg_go.go#L18-L21

File: ddtrace/tracer/tracer.go [hunk 1]
```diff
@@ -162,6 +162,7 @@ func newUnstartedTracer(opts ...StartOption) *tracer {
 func newTracer(opts ...StartOption) *tracer {
 	t := newUnstartedTracer(opts...)
 	c := t.config
+	log.Debug("Tracer starting. %#v", c)
```
Comment (knusbaum) [inline ddtrace/tracer/tracer.go:165]: I think we can remove this log. We already have startup logging [here](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/tracer.go#L97).

File: contrib/julienschmidt/httprouter/httprouter.go [hunk 2]
```diff
@@ -35,6 +36,7 @@ func New(opts ...RouterOption) *Router {
 		cfg.spanOpts = append(cfg.spanOpts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
 	}
 	cfg.spanOpts = append(cfg.spanOpts, tracer.Measured())
+	log.Debug("julienschmidt/httprouter: Configuring Router: %#v", cfg)
```
Comment (knusbaum) [inline contrib/julienschmidt/httprouter/httprouter.go:39]: Let's add `contrib/` to the path here for consistency.

File: contrib/gopkg.in/jinzhu/gorm.v1/gorm.go [hunk 3]
```diff
@@ -67,6 +67,7 @@ func WithCallbacks(db *gorm.DB, opts ...Option) *gorm.DB {
 	for _, fn := range opts {
 		fn(cfg)
 	}
+	log.Debug("gopkg.in/jinzhu/gorm.v1: Registering Callbacks: %#v", cfg)
```
Comment (knusbaum) [inline contrib/gopkg.in/jinzhu/gorm.v1/gorm.go:70]: Let's add `contrib/` to the path here as well.

File: contrib/gofiber/fiber.v2/fiber.go [hunk 4]
```diff
@@ -26,6 +27,7 @@ func Middleware(opts ...Option) func(c *fiber.Ctx) error {
 	for _, fn := range opts {
 		fn(cfg)
 	}
+	log.Debug("gofiber/fiber.v2: Middleware: %#v", cfg)
```
Comment (knusbaum) [inline contrib/gofiber/fiber.v2/fiber.go:30]: `contrib/` should be added here.

File: contrib/go-pg/pg.v10/pg_go.go [hunk 5]
```diff
@@ -11,12 +11,14 @@ import (
...
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
 
 	"github.com/go-pg/pg/v10"
 )
 
 // Wrap augments the given DB with tracing.
 func Wrap(db *pg.DB) {
+	log.Debug("go-pg/pg.v10: Wrapping Database")
```
Comment (knusbaum) [inline contrib/go-pg/pg.v10/pg_go.go:21]: `contrib/` here too.

File: ddtrace/tracer/tracer.go [hunk 6]
```diff
@@ -322,6 +323,7 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 		// this is a brand new trace, sample it
 		t.sample(span)
 	}
+	log.Debug("Started Span: %v, Operation: %s, Resource: %s, Tags: %v, %v", span, span.Name, span.Resource, span.Meta, span.Metrics)
```
Comment (knusbaum) [inline ddtrace/tracer/tracer.go:326]: This format is complicated and it isn't clear what exactly the output will be like.
We should add a test to `ddtrace/tracer/log_test.go` that tests this output to ensure the format continues to make sense in the future.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Looks pretty good. Just a couple minor things.

The largest is that we should add a test for the span creation log.

All integration logs should have the full path starting with `contrib/` for consistency. I marked all the ones I saw with comments.

**Review by gbbr — APPROVED** [review]
LGTM

**Comment by nicoledanuwidjaja** [comment]
@felixge Put up a PR to fix it. Thanks!
https://github.com/DataDog/dd-trace-go/pull/840

**Comment by felixge** [comment]
Thanks @nicoledanuwidjaja : )

**Pull request merged**